src/text.rs

```diff
@@ -373,15 +386,15 @@ impl TextContextImpl {
         self.fonts.get_mut(id.0)
     }
 
-    pub fn find_font<F, T>(&mut self, paint: &Paint, mut callback: F) -> Result<T, ErrorKind>
+    pub fn find_font<F, T>(&mut self, font_ids: [Option<FontId>; 8], mut callback: F) -> Result<T, ErrorKind>
```

Comment:
I think just passing a reference to a slice would be sufficient here, as well as in many other places.

Reply:
Mainly because I copied it from the type from struct.

Reply:
I store font ids as `Vec` now, not sure if this is a good idea.
I also used `Rc` to avoid too much copying.
At least when building the cache, there's still a lot of cloning required.

Not sure if it's better than the suggested change?